### PR TITLE
Spline feed update

### PR DESCRIPTION
**Pull request by mcsloy**
This pull request brings the `CubicSpline` class into line with the PyTorch Module.

### CODE REVIEW

File: tbmalt/common/maths/interpolation.py [hunk 1]
```diff
@@ -237,6 +241,9 @@ class PolyInterpU:
 
     """
 
+    # TODO: Resolve bug causing incorrect value to be returned when
```
Comment (WbSun723) [inline tbmalt/common/maths/interpolation.py:244]: Hi Adam, is there an example or unittest to reproduce this bug? I will also have a look on this issue.

File: tests/unittests/test_dftb_feeds.py [hunk 2]
```diff
@@ -146,6 +146,120 @@ def test_scipyskfeed_batch(device, skf_file: str):
 # Note that gradient tests are not performed on the ScipySkFeed as it is not
 # backpropagatable due to its use of Scipy splines for interpolation.
 
+def test_skfeed_poly_single(device, skf_file: str):
+    """ScipySkFeed matrix single system operability tolerance test"""
```
Comment (WbSun723) [inline tests/unittests/test_dftb_feeds.py:150]: Hi Adam, might be SKFeed here.

File: tests/unittests/test_dftb_feeds.py [hunk 3]
```diff
@@ -146,6 +146,120 @@ def test_scipyskfeed_batch(device, skf_file: str):
...
+
+    if device.type == "cuda":
+        pytest.skip("Scipy splines do not support CUDA.")
```
Comment (WbSun723) [inline tests/unittests/test_dftb_feeds.py:153]: As this is the test for SKFeed, CUDA should also be supported.

File: tests/unittests/test_dftb_feeds.py [hunk 4]
```diff
@@ -552,3 +666,18 @@ def test_hubbardfeed_batch(device, skf_file):
     check_2 = torch.allclose(predicted, reference)
     assert check_2, 'Predicted hubbard value errors exceed allowed tolerance'
 
+
+if __name__ == '__main__':
```
Comment (WbSun723) [inline tests/unittests/test_dftb_feeds.py:670]: Hi Adam, seems these codes are only for debugging :)

File: tests/unittests/test_dftb_feeds.py (hunk 2 above)
Comment (mcsloy) [reply tests/unittests/test_dftb_feeds.py]: Changed "ScipySkFeed" to "SkFeed" in the relevant test methods.

File: tests/unittests/test_dftb_feeds.py (hunk 4 above)
Comment (mcsloy) [reply tests/unittests/test_dftb_feeds.py]: Removed debugging code block.

File: tests/unittests/test_dftb_feeds.py (hunk 3 above)
Comment (mcsloy) [reply tests/unittests/test_dftb_feeds.py]: Tests performed on non-Scipy feeds will no longer be skipped when running on the GPU.

File: tbmalt/common/maths/interpolation.py [hunk 5]
```diff
@@ -450,16 +467,70 @@ class CubicSpline(torch.nn.Module):
         >>> torch.sin(torch.tensor([3.5]))
         >>> tensor([-0.3508])
```
[the remark this answers is not in the document]
Comment (mcsloy) [reply tbmalt/common/maths/interpolation.py:468]: Good catch here, I completely missed this one. Updated docstring examples to bring them into line with current semantics.

File: tbmalt/common/maths/interpolation.py (hunk 1 above)
Comment (mcsloy) [reply tbmalt/common/maths/interpolation.py:244]: Although this example is rather contrived it does demonstrate how non-trivial the error can become under certain situations:
```python
from tbmalt.common.maths.interpolation import PolyInterpU
import torch

x = torch.linspace(0, 10, 10)
y_reference = torch.rand(10)
poly = PolyInterpU(x, y_reference)
y_predicted = poly(x)
y_delta = y_predicted - y_reference
```
The error becomes most pronounced at location of the last data-point. In this example the predicted value can be up to ±60 off from the reference value which is only ever in the range 0.0-1.0.

**Pull request merged**